src/middlewares/enricher.ts

```diff
@@ -0,0 +1,52 @@
+import { Context, Next } from 'koa';
+import {
+  ProcessorTransformationRequest,
+  RouterTransformationRequest,
+  RouterTransformationRequestData,
+} from '../types';
+import GeoLocationHelper from '../helpers/geoLocation';
+
+export type DTRequest = RouterTransformationRequest | ProcessorTransformationRequest[];
+
+export default class Enricher {
```

Comment:
I would name the class as `GeoEnricher` and the function `enrichGeoLocation` as `enrich` only.

Reply:
Makes sense. Will update

Reply:
Updated